typing/jkind.ml

```diff
@@ -1250,49 +1277,65 @@ module Const = struct
         name : string
       }
 
-    let mk_jkind ~mode_crossing ~nullability (layout : Layout.Const.t) =
+    let mk_jkind ~mode_crossing ~nullability ~separability
+        (layout : Layout.Const.t) =
       let mod_bounds =
         (match mode_crossing with
         | true -> Mod_bounds.min
         | false -> Mod_bounds.max)
         |> Mod_bounds.set_nullability nullability
+        |> Mod_bounds.set_separability separability
       in
       { layout; mod_bounds; with_bounds = No_with_bounds }
 
     let any =
-      { jkind = mk_jkind Any ~mode_crossing:false ~nullability:Maybe_null;
+      { jkind =
+          mk_jkind Any ~mode_crossing:false ~nullability:Maybe_null
+            ~separability:Non_separable;
         name = "any"
       }
 
     let any_mod_everything =
-      { jkind = mk_jkind Any ~mode_crossing:true ~nullability:Maybe_null;
+      { jkind =
+          mk_jkind Any ~mode_crossing:true ~nullability:Maybe_null
+            ~separability:Non_separable;
```

Comment:
This disagrees with the meaning of `everything` in the documentation. I'm not sure what's better, but we need to be consistent. Do you have a sense for what we should do?

Reply:
I think separability should be consistent with nullability, and we'll need to update the docs. `everything` really means everything modal.

Reply:
Then sadly `immediate` won't be `value mod everything`. Instead it would be `value mod everything non_null non_float`. Seems a little sad. But if we're going to go with this, maybe update the doc in this PR?

Reply:
`value mod everything non_float` since `value` is already `non_null`.

I'll update the docs in a subsequent PR now. I've noticed that they get a few things wrong even without separability, I'll fix that too.

Reply:
OK